js/mixins/uploader.js

```diff
@@ -1,195 +1,210 @@
+    minWidthImageError: i18n._('Image is not wide enough.'),
+    minSizeError: i18n._('{file} is too small, minimum file size is {minSizeLimit}.', {interpolation: interpolation}),
+    noFilesError: i18n._('No files to upload.'),
+    onLeave: i18n._('The files are being uploaded, if you leave now the upload will be canceled.'),
+    retryFailTooManyItemsError: i18n._('Retry failed - you have reached your file limit.'),
+    sizeError: i18n._('{file} is too large, maximum file size is {sizeLimit}.', {interpolation: interpolation}),
+    tooManyItemsError: i18n._('Too many items ({netItems}) would be uploaded. Item limit is {itemLimit}.', {interpolation: interpolation}),
+    typeError: i18n._('{file} has an invalid extension. Valid extension(s): {extensions}.', {interpolation: interpolation}),
+    unsupportedBrowserIos8Safari: i18n._('Unrecoverable error - this browser does not permit file uploading of any kind due to serious bugs in iOS8 Safari. Please use iOS8 Chrome until Apple fixes these issues.'),
+};
 
+/**
+ * This is a mixin handling the fine-uploader logic.
+ *
+ * See the official documentation for more details:
+ * http://docs.fineuploader.com/
+ */
+export default {
+    data() {
+        return {
+            dropping: false,
+            upload_endpoint: null,
+            HAS_FILE_API: HAS_FILE_API,
+            files: []
+        };
+    },
+    ready() {
+        this.$uploader = new qq.FineUploaderBasic({
+            debug: DEBUG,
+            multiple: this.$options.hasOwnProperty('upload_multiple') ? this.$options.upload_multiple : true,
+            uploaderType: 'basic',
+            autoUpload: this.$options.hasOwnProperty('autoUpload') ? this.$options.autoUpload : true,
+            button: this.$els.uploadBtn,
+            request: {
+                endpoint: this.upload_endpoint,
+                inputName: 'file'
+            },
+            callbacks: {
+                onUpload: this.on_upload,
+                onSubmitted: this.on_submit,
+                onProgress: this.on_progress,
+                onComplete: this.on_complete,
+                onError: this.on_error,
+            },
+            messages: messages,
+            validation: {
+                allowedExtensions: [
+                    // Base
+                    'csv', 'txt', 'json', 'pdf', 'xml', 'rdf', 'rtf',
+                    // OpenOffice
+                    'ods', 'odt', 'odp', 'odg',
+                    // Microsoft Office
+                    'xls', 'xlsx', 'doc', 'docx', 'pps', 'ppt',
+                    // Archives
+                    'tar', 'gz', 'tgz', 'rar', 'zip', '7z', 'xz', 'bz2',
+                    // Images
+                    'jpeg', 'jpg', 'jpe', 'gif', 'png', 'dwg', 'svg', 'tiff', 'ecw', 'svgz',
+                    // Geo
+                    'shp', 'kml', 'kmz', 'gpx', 'shx', 'ovr',
+                    // Misc
+                    'dbf', 'prj', 'sql', 'epub', 'sbn', 'sbx', 'cpg', 'lyr', 'xsd', 'owl'
```

Comment:
According to https://developers.google.com/transit/gtfs/examples/gtfs-feed, GTFS is a format but not a file extension. A GTFS should be stored in a .txt file.